cdap-app-fabric/src/main/java/co/cask/cdap/internal/app/services/ApplicationLifecycleService.java

```diff
@@ -396,13 +354,19 @@ public void upgrade(boolean continueOnFailure) throws Exception {
           continue;
         }
 
+        // if this was etlbatch or etlrealtime template
+        if ("ETLBatch".equals(appSpec.getName()) || "ETLRealtime".equals(appSpec.getName())) {
```

Comment:
though this logic could be moved into adapter service, where we would know it's a template.  Will do that.